File: TI1.1 SOAP_JAVA_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Java_GR06/src/java/ec/edu/monster/pruebas/PruebaConversorCompleto.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.pruebas;

import ec.edu.monster.services.ConversorTemperatura;
import ec.edu.monster.services.ConversorLongitud;
import ec.edu.monster.services.ConversorMasa;
import ec.edu.monster.services.AutenticacionService;
import ec.edu.monster.modelo.Usuario;
/**
 *
 * @author josue
 */
public class PruebaConversorCompleto {
    
    public static void main(String[] args) {
        pruebaAutenticacion();
        pruebaTemperatura();
        pruebaLongitud();
        pruebaMasa();
    }
    
    private static void pruebaAutenticacion() {
        System.out.println("╔══════════════════════════════════════╗");
        System.out.println("║  PRUEBAS DE AUTENTICACIÓN            ║");
        System.out.println("╚══════════════════════════════════════╝\n");
        
        // Prueba login exitoso
        Usuario usuario1 = AutenticacionService.autenticar("MONSTER", "MONSTER9");
        System.out.println("✓ Login correcto: " + (usuario1 != null ? "✓ EXITOSO" : "✗ FALLÓ"));
        
        // Prueba login fallido
        Usuario usuario2 = AutenticacionService.autenticar("WRONG", "WRONG");
        System.out.println("✓ Login incorrecto rechazado: " + (usuario2 == null ? "✓ EXITOSO" : "✗ FALLÓ"));
        
        System.out.println();
    }
    
    private static void pruebaTemperatura() {
        System.out.println("╔══════════════════════════════════════╗");
        System.out.println("║  PRUEBAS DE TEMPERATURA              ║");
        System.out.println("╚══════════════════════════════════════╝\n");
        
        ConversorTemperatura conv = new ConversorTemperatura();
        
        // Celsius a Fahrenheit
        double c2f = conv.celsiusAFahrenheit(0);
        System.out.println("0°C → " + c2f + "°F " + (c2f == 32 ? "✓" : "✗"));
        
        // Celsius a Kelvin
        double c2k = conv.celsiusAKelvin(0);
        System.out.println("0°C → " + c2k + "K " + (c2k == 273.15 ? "✓" : "✗"));
        
        // Fahrenheit a Celsius
        double f2c = conv.fahrenheitACelsius(212);
        System.out.println("212°F → " + f2c + "°C " + (f2c == 100 ? "✓" : "✗"));
        
        System.out.println();
    }
    
    private static void pruebaLongitud() {
        System.out.println("╔══════════════════════════════════════╗");
        System.out.println("║  PRUEBAS DE LONGITUD                 ║");
        System.out.println("╚══════════════════════════════════════╝\n");
        
        ConversorLongitud conv = new ConversorLongitud();
        
        // Metro a Kilómetro
        double m2km = conv.metroAKilometro(1000);
        System.out.println("1000 m → " + m2km + " km " + (m2km == 1.0 ? "✓" : "✗"));
        
        // Kilómetro a Metro
        double km2m = conv.kilometroAMetro(1);
        System.out.println("1 km → " + km2m + " m " + (km2m == 1000.0 ? "✓" : "✗"));
        
        // Metro a Milla
        double m2mi = conv.metroAMilla(1609.344);
        System.out.println("1609.344 m → " + String.format("%.2f", m2mi) + " mi " + 
                          (Math.abs(m2mi - 1.0) < 0.01 ? "✓" : "✗"));
        
        System.out.println();
    }
    
    private static void pruebaMasa() {
        System.out.println("╔══════════════════════════════════════╗");
        System.out.println("║  PRUEBAS DE MASA                     ║");
        System.out.println("╚══════════════════════════════════════╝\n");
        
        ConversorMasa conv = new ConversorMasa();
        
        // Kilogramo a Gramo
        double kg2g = conv.kilogramoAGramo(1);
        System.out.println("1 kg → " + kg2g + " g " + (kg2g == 1000.0 ? "✓" : "✗"));
        
        // Gramo a Kilogramo
        double g2kg = conv.gramoAKilogramo(1000);
        System.out.println("1000 g → " + g2kg + " kg " + (g2kg == 1.0 ? "✓" : "✗"));
        
        // Kilogramo a Libra
        double kg2lb = conv.kilogramoALibra(1);
        System.out.println("1 kg → " + String.format("%.2f", kg2lb) + " lb " + 
                          (Math.abs(kg2lb - 2.20462) < 0.01 ? "✓" : "✗"));
        
        System.out.println();
    }
}

File: Deberes_PP_Arquitectura-entregable/TI1.2 SOAP_DOTNET_SINBDD_GR06/01 SERVIDOR/ConUni_Soap_Dotnet_GR06/ConUni_Soap_Dotnet_GR06/Pruebas/PruebaConversorCompleto.cs
using ConUni_Soap_Dotnet_GR06.Servicios;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace ConUni_Soap_Dotnet_GR06.Pruebas
{
    /// <summary>
    /// Clase para probar todos los conversores y autenticación
    /// </summary>
    public class PruebaConversorCompleto
    {
        public static void Main(string[] args)
        {
            PruebaAutenticacion();
            PruebaTemperatura();
            PruebaLongitud();
            PruebaMasa();
        }

        private static void PruebaAutenticacion()
        {
            Console.WriteLine("╔══════════════════════════════════════╗");
            Console.WriteLine("║  PRUEBAS DE AUTENTICACIÓN            ║");
            Console.WriteLine("╚══════════════════════════════════════╝\n");

            // Prueba login exitoso
            var usuario1 = AutenticacionService.Autenticar("MONSTER", "MONSTER9");
            Console.WriteLine($"✓ Login correcto: {(usuario1 != null ? "✓ EXITOSO" : "✗ FALLÓ")}");

            // Prueba login fallido
            var usuario2 = AutenticacionService.Autenticar("WRONG", "WRONG");
            Console.WriteLine($"✓ Login incorrecto rechazado: {(usuario2 == null ? "✓ EXITOSO" : "✗ FALLÓ")}");

            Console.WriteLine();
        }

        private static void PruebaTemperatura()
        {
            Console.WriteLine("╔══════════════════════════════════════╗");
            Console.WriteLine("║  PRUEBAS DE TEMPERATURA              ║");
            Console.WriteLine("╚══════════════════════════════════════╝\n");

            var conv = new ConversorTemperatura();

            // Celsius a Fahrenheit
            double c2f = conv.CelsiusAFahrenheit(0);
            Console.WriteLine($"0°C → {c2f}°F {(c2f == 32 ? "✓" : "✗")}");

            // Celsius a Kelvin
            double c2k = conv.CelsiusAKelvin(0);
            Console.WriteLine($"0°C → {c2k}K {(c2k == 273.15 ? "✓" : "✗")}");

            // Fahrenheit a Celsius
            double f2c = conv.FahrenheitACelsius(212);
            Console.WriteLine($"212°F → {f2c}°C {(f2c == 100 ? "✓" : "✗")}");

            Console.WriteLine();
        }

        private static void PruebaLongitud()
        {
            Console.WriteLine("╔══════════════════════════════════════╗");
            Console.WriteLine("║  PRUEBAS DE LONGITUD                 ║");
            Console.WriteLine("╚══════════════════════════════════════╝\n");

            var conv = new ConversorLongitud();

            // Metro a Kilómetro
            double m2km = conv.MetroAKilometro(1000);
            Console.WriteLine($"1000 m → {m2km} km {(m2km == 1.0 ? "✓" : "✗")}");

            // Kilómetro a Metro
            double km2m = conv.KilometroAMetro(1);
            Console.WriteLine($"1 km → {km2m} m {(km2m == 1000.0 ? "✓" : "✗")}");

            // Metro a Milla
            double m2mi = conv.MetroAMilla(1609.344);
            Console.WriteLine($"1609.344 m → {m2mi:F2} mi {(Math.Abs(m2mi - 1.0) < 0.01 ? "✓" : "✗")}");

            Console.WriteLine();
        }

        private static void PruebaMasa()
        {
            Console.WriteLine("╔══════════════════════════════════════╗");
            Console.WriteLine("║  PRUEBAS DE MASA                     ║");
            Console.WriteLine("╚══════════════════════════════════════╝\n");

            var conv = new ConversorMasa();

            // Kilogramo a Gramo
            double kg2g = conv.KilogramoAGramo(1);
            Console.WriteLine($"1 kg → {kg2g} g {(kg2g == 1000.0 ? "✓" : "✗")}");

            // Gramo a Kilogramo
            double g2kg = conv.GramoAKilogramo(1000);
            Console.WriteLine($"1000 g → {g2kg} kg {(g2kg == 1.0 ? "✓" : "✗")}");

            // Kilogramo a Libra
            double kg2lb = conv.KilogramoALibra(1);
            Console.WriteLine($"1 kg → {kg2lb:F2} lb {(Math.Abs(kg2lb - 2.20462) < 0.01 ? "✓" : "✗")}");

            Console.WriteLine();
        }
    }
}